### DISCUSSION TITLE
Can't get Google Analytics to work properly

### DISCUSSION THREAD

**Discussion by Devalo** [vercel/next.js#17447] [Help] [answered]
Hi,
I've been looking around after a proper solution for getting Google Analytics to work with Next.js, 
without any luck. It doesn't seem like Google Analytics updates the viewed page when I'm navigating around
my website. If I enter a direct URL, or refresh, it will register the page as it should.

All my internal urls are using the Link component. Like this:

```
<Link href="/some-url">
  <a>Visit some site</a>
</Link>
```

Google Analytics doesn't register when I'm clicking the link.
If I'm moving directly to the page:
`www.somepage.com/some-url`
It gets registered 

I have a layout component as per next tutorial. I've tried adding the analytics script within the Head component in here.
I've also tried adding the analytics script to every page, within the Head component

```
<Head>
    <script async src="https://www.googletagmanager.com/gtag/js?id=IDNUMBER"></script>
    <script dangerouslySetInnerHTML={{
        __html: `
          window.dataLayer = window.dataLayer || [];
          function gtag(){dataLayer.push(arguments);}
          gtag('js', new Date());
          
          gtag('config', 'UA-179245515-1');
      `,
    }} />
</Head>
```

No matter where I put this script, it doesn't seem to register the page clicks correctly.
I suppose this has to do with the way Link is built. Since it doesn't do a complete refresh.

How can I install Google Analytics the proper way?

**Comment by jamesmosier** [comment, marked as answer]
Is `router.events.on('routeChangeComplete'` not firing for you in `_app.js`? Maybe try to put a `console.log` on [line 9](https://github.com/vercel/next.js/blob/c731c636315ddfa289bbc4b50b73c741c7978a1a/examples/with-google-analytics/pages/_app.js#L9) to see what event may or may not be being fired off.

**Reply by Devalo** [reply]
You know what, after adding it I got it working. I was looking for a way to delete this entry before you answered. Thanks for your help. This thread may be closed :-)
